src/txn/txn_rollback_to_stable.c

```diff
@@ -274,38 +327,49 @@ __rollback_row_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW
 
         /*
          * Stop processing when we find the newer version value of this key is stable according to
-         * the current version stop timestamp when it is not appending the selected update to the
-         * update chain. Also it confirms that history store doesn't contains any newer version than
-         * the current version for the key.
+         * the current version stop timestamp and transaction id when it is not appending the
+         * selected update to the update chain. Also it confirms that history store doesn't contains
+         * any newer version than the current version for the key.
          */
-        if (!replace && hs_stop_durable_ts <= rollback_timestamp) {
+        if (!replace &&
+          (!__rollback_check_if_txnid_non_committed(session, cbt->upd_value->tw.stop_txn)) &&
+          (hs_stop_durable_ts <= rollback_timestamp)) {
             __wt_verbose(session, WT_VERB_RECOVERY_RTS(session),
-              "history store update valid with stop timestamp: %s and stable timestamp: %s",
+              "history store update valid with stop timestamp: %s, stable timestamp: %s and txnid "
+              ": %" PRIu64,
```

Comment:
Combine it with the previous line.

Reply:
I tried combining it but s_all splits it into two lines,